### PR TITLE
Improve guides - Closes #1445

### LINKED ISSUE #1445: Guide improvements
#### Guides for delegates
- It would be nice if we could also share the process to create an account and register it as a delegate at the beginning of these instructions
- describe the process to create an account and register it as a delegate at the beginning of these instructions.
- Implicit assumption that the user ‘lisk’ exists on the user’s system and that ‘lisk-core’ is cloned to the home directory.
- Missing information on how to create, configure and use ‘custom-config.json’.

#### Guides for integrators
- It would also be nice to include references to the available transaction asset schemas such as. It is required to create transactions with the lisk-core cli using the ‘--asset’ property:
https://liskhq.github.io/lisk-docs/lisk-sdk/modules/token-module.html#transactions
https://liskhq.github.io/lisk-docs/lisk-sdk/modules/keys-module.html#transactions
https://liskhq.github.io/lisk-docs/lisk-sdk/modules/dpos-module.html#transactions

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #1445

### CODE REVIEW

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 1]
```diff
@@ -108,66 +175,126 @@ The new delegate name should be under `dpos.delegate.username`:
...
+[source,bash]
+----
+cp config/default/config.json config/default/custom-config.json
+----
+--
+====
+
+=== Option 1: via application CLI
+
 If the blockchain application was xref:{url_guides_setup_bootstrap}[bootstrapped with Lisk Commander], it is possible to generate the relevant forging data conveniently with the command xref:{url_references_cli_forgingconfig}[forging:config].
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
If the blockchain application was xref:{url_guides_setup_bootstrap}[bootstrapped with Lisk Commander], it is possible to conveniently generate the relevant forging data using the command xref:{url_references_cli_forgingconfig}[forging:config].
```
```suggestion
If the blockchain application was xref:{url_guides_setup_bootstrap}[bootstrapped with Lisk Commander], it is possible to generate the relevant forging data conveniently with the command xref:{url_references_cli_forgingconfig}[forging:config].
```

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 2]
```diff
@@ -232,49 +359,184 @@ lisk hash-onion --json
...
+[source,bash]
+----
+TEMP_FILE=$( mktemp )
+jq --slurp '.[0] * .[1]' custom-config.json custom-forging.json >$TEMP_FILE
+mv $TEMP_FILE custom-config.json
+----
+
+Open `custom-config.json` to verify it was updated correctly.
+
+Beside adding the delegate config data, the other options in the config can be adjusted as desired.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
Besides adding the delegate config data, the other options in the config can be adjusted as desired.
```
```suggestion
Beside adding the delegate config data, the other options in the config can be adjusted as desired.
```

Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
Besides adding the delegate config data, the other options in the config can be adjusted as desired.
```
```suggestion
Beside adding the delegate config data, the other options in the config can be adjusted as desired.
```

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 3]
```diff
@@ -232,49 +359,184 @@ lisk hash-onion --json
...
+Find examples of config options that might be interesting to update right below.
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
Examples of config options that might be interesting to update can be found below.
```
```suggestion
Find examples of config options that might be interesting to update right below.
```

Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
Examples of config options that might be interesting to update can be found below.
```
```suggestion
Find examples of config options that might be interesting to update right below.
```

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 4]
```diff
@@ -232,49 +359,184 @@ lisk hash-onion --json
...
+. If forging is enabled for a delegate for the **first time**, use `0` as the value for `HEIGHT`, `MAXHEIGHTPREVIOUSLYFORGED`, and `MAXHEIGHTPREVOTED`.
+Afterwards, it is always required to use the latest values of the `forger_info` info data.
+. To avoid being xref:{url_sdk_protocol_puninshment}[punished] by the network, ensure to adhere to the following points:
+.. Never use outdated <<forger_info-data>>.
+.. Never activate forging for the same delegate on two or more nodes at the same time, (a.k.a double-forging).
 ====
 
+=== via application CLI
+
+If the blockchain application was xref:{url_guides_setup_bootstrap}[bootstrapped with Lisk Commander], it is possible to generate the relevant forging data conveniently with the command xref:{url_references_cli_forgingenable}[forging:enable].
```
Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
If the blockchain application was xref:{url_guides_setup_bootstrap}[bootstrapped with Lisk Commander], it is possible to conveniently generate the relevant forging data using the command xref:{url_references_cli_forgingenable}[forging:enable].
```
```suggestion
If the blockchain application was xref:{url_guides_setup_bootstrap}[bootstrapped with Lisk Commander], it is possible to generate the relevant forging data conveniently with the command xref:{url_references_cli_forgingenable}[forging:enable].
```

Comment (chris529) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
If the blockchain application was xref:{url_guides_setup_bootstrap}[bootstrapped with Lisk Commander], it is possible to conveniently generate the relevant forging data using the command xref:{url_references_cli_forgingenable}[forging:enable].
```
```suggestion
If the blockchain application was xref:{url_guides_setup_bootstrap}[bootstrapped with Lisk Commander], it is possible to generate the relevant forging data conveniently with the command xref:{url_references_cli_forgingenable}[forging:enable].
```

**Review by chris529 — COMMENTED** [review]
Looks good, just a few minor points 👍

File: docs/modules/ROOT/pages/integrate-blockchain/create-send-transactions.adoc [hunk 5]
```diff
@@ -14,9 +14,25 @@
...
+:url_sdk_token_module_txs: lisk-sdk::modules/token-module.adoc#transactions
+:url_sdk_keys_module_txs: lisk-sdk::modules/keys-module.adoc#transactions
+:url_sdk_dpos_module_txs: lisk-sdk::modules/dpos-module.adoc#transactions
+:url_protocol_transactions: understand-blockchain/lisk-protocol/transactions.adoc#types
 
 This guide describes different methods of creating and sending transactions to Lisk nodes.
 
+== Default transactions
+
+The default modules already come with a set of xref:{url_protocol_transactions}[default transactions], which can be sent to the blockchain application without writing any custom code.
```
Comment (sameersubudhi) [inline docs/modules/ROOT/pages/integrate-blockchain/create-send-transactions.adoc]:
```suggestion
The default modules already come with a set of xref:{url_protocol_transactions}[default transactions] out of the box, which can be sent to the blockchain application without writing any custom code.
```

File: docs/modules/ROOT/pages/integrate-blockchain/create-send-transactions.adoc [hunk 6]
```diff
@@ -14,9 +14,25 @@
...
+
+Each transaction type is defined in a module.
+For more information about the different modules and their transaction assets, check out the respective module references:
+
+.Transactions of...
+* xref:{url_sdk_token_module_txs}[... the Token module]
+* xref:{url_sdk_keys_module_txs}[... the Keys module]
+* xref:{url_sdk_dpos_module_txs}[... the DPoS module]
```
Comment (sameersubudhi) [inline docs/modules/ROOT/pages/integrate-blockchain/create-send-transactions.adoc]: Do we need the ellipsis at the beginning of each point?
```suggestion
.Transactions of...
* xref:{url_sdk_token_module_txs}[the Token module]
* xref:{url_sdk_keys_module_txs}[the Keys module]
* xref:{url_sdk_dpos_module_txs}[the DPoS module]
```

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 7]
```diff
@@ -36,42 +38,107 @@ For more information about forging, please check the xref:{url_explanations_cons
 Before it is possible to enable forging on a node, it is required to register as a delegate.
 This is achieved by choosing a unique delegate name, and posting a corresponding xref:{url_protocol_tx_registerdelegegate}[Register Delegate] transaction to the network.
 
-There are many possibilities to post the transaction, use the one you prefer: {url_lisk_wallet}[Lisk Desktop^], xref:{url_elements}[Lisk Elements], or the xref:{url_framework_dashbobardplugin}[].
+TIP: There are many possibilities to post the transaction, use the one you prefer: {url_lisk_wallet}[Lisk Desktop^], xref:{url_elements}[Lisk Elements], or the xref:{url_framework_dashbobardplugin}[].
 
 One option is to create the transaction object with the application CLI:
```
Comment (sameersubudhi) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
An option is to create the transaction object with the application CLI:
```

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 8]
```diff
@@ -36,42 +38,107 @@ For more information about forging, please check the xref:{url_explanations_cons
...
+--
 [source,bash]
 ----
- ./bin/run account:get ed86183c22f8399f7aa28f7d2c2f1680224f7281 --pretty
+./bin/run account:get ed86183c22f8399f7aa28f7d2c2f1680224f7281 --pretty
 ----
+--
+=====
 
 The new delegate name should be under `dpos.delegate.username`:
```
Comment (sameersubudhi) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
The new delegate name should be available under `dpos.delegate.username`:
```
or
```suggestion
The new delegate name should be displayed under `dpos.delegate.username`:
```

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 9]
```diff
@@ -108,66 +175,126 @@ The new delegate name should be under `dpos.delegate.username`:
...
 
-[[config_forging_data]]
-== Add forging data to config
+IMPORTANT: This step only needs to be performed when a delegate is enabling forging on a node for the first time.
 
-To enable your node to forge for a particular delegate, firstly it is required to insert some data into the config file under the `forging.delegates` array:
+To enable your node to forge for a particular delegate, firstly it is required to insert some data into the config file under the `forging.delegates` array as described below:
 
-* `publicKey`: The publicKey of the delegate.
+* `address`: The address of the delegate.
```
Comment (sameersubudhi) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]: Should we also state that the address should be in hex format?

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 10]
```diff
@@ -108,66 +175,126 @@ The new delegate name should be under `dpos.delegate.username`:
...
+The command will ask for the delegate passphrase, and in order to symmetrically encrypt the passphrase for the config, it will also ask for a password.
 
-For this example, we reduce the value of `count` to  4000, with a default distance of `1000` this will only produce 5 hashes in the hash onion.
-Generally though, it is recommended to use high values for `count`, so the delegate has enough hashes to forge for a long time.
+IMPORTANT: Store the password you used for the encryption somewhere safe.
+It will be required everytime you wish to enable forging, in order to decrypt the delegates' passphrase in the config.
 
-.forging_data.json
-[source,json]
+After providing the required inputs, the delegate configuration data will be saved in the file `forging_config.json`.
```
Comment (sameersubudhi) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]: Please rename the file for consistency purposes.
```suggestion
After providing the required inputs, the delegate configuration data will be saved in the file `delegate_config.json`.
```

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 11]
```diff
@@ -108,66 +175,126 @@ The new delegate name should be under `dpos.delegate.username`:
...
-Option B - Creating the data with Lisk Commander::
-+
---
+
+<1>  The list of delegates who are allowed to forge on this node.
+====
+
+=== Option 2: with Lisk Commander
+
 Please ensure the xref:{url_commander_commands}[Lisk Commander] is installed in a secure environment.
```
Comment (sameersubudhi) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
Please ensure that the xref:{url_commander_commands}[Lisk Commander] is installed in a secure environment.
```

Please also consider setting the numbers in a single column in the following [bash instructions](https://liskhq.github.io/lisk-docs/run-blockchain/forging.html#option-2-with-lisk-commander):
```
$ lisk
lisk passphrase:encrypt --json
Please enter your secret passphrase: *****       (1)
Please re-enter your secret passphrase: *****
Please enter your password: ***                  (2)
Please re-enter your password: ***
{
        "encryptedPassphrase": "iterations=1000000&cipherText=30a3c8&iv=b0d7322bf24e0dfe08462f4f&salt=aa7e26c9f4317b61b4f45b5c6909f941&tag=a2e0eadaf1f11a10b342965bc3bafc68&version=1",
}
```

Also, please correct the typo (`encryptedPassphraser`) in the statement below:
```
Add the hash onion and the delegate address in hexadecimal representation to the object with the `encryptedPassphraser`.
```

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 12]
```diff
@@ -232,49 +359,232 @@ lisk hash-onion --json
...
+Add the JSON object to the config file of the application.
+
+`forging.delegates` stores the list of delegates who are allowed to forge on this node:
+
+Merge the delegate config with the custom config which was created in <<creating-a-custom-config-file,step 2.1>> above, to add the delegate information to the application configuration:
+
+[source,bash]
+----
+TEMP_FILE=$( mktemp )
+jq --slurp '.[0] * .[1]' custom-config.json delegate_config.json >$TEMP_FILE
```
Comment (sameersubudhi) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]:
```suggestion
jq --slurp '.[0] * .[1]' custom-config.json delegate_config.json > $TEMP_FILE
```

File: docs/modules/ROOT/pages/run-blockchain/forging.adoc [hunk 13]
```diff
@@ -108,66 +175,126 @@ The new delegate name should be under `dpos.delegate.username`:
...
+* `address`: The address of the delegate.
 * `encryptedPassphrase`: The symmetrically encrypted 12 word mnemonic passphrase of the delegate account.
 * `hashOnion`: The hash onion stores the random seeds of the delegate for each forging round.
 
-To create the forging configuration data, it is recommended to use one of the following alternatives listed below:
+To create the delegate configuration data,
+
+. first create a custom config file
+. then use one of the following alternatives to generate the delegate config data: <<via-application-cli>> or <<with-lisk-commander>>
+. lastly, add the delegate config data to the custom config file.
```
Comment (sameersubudhi) [inline docs/modules/ROOT/pages/run-blockchain/forging.adoc]: Please consider starting the statements in the list with a capital letter, to maintain consistency across the document.

Also, currently, the second statement in the list ends with `[with-lisk-commander]`. Please consider updating it to display like `with Lisk Commander`, for visual aesthetics.

```suggestion
To create the delegate configuration data,

. First, create a custom config file
. Then use one of the following alternatives to generate the delegate config data: <<via-application-cli>> or <<with-lisk-commander>>
. Lastly, add the delegate config data to the custom config file.
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please find my observations/suggestions below.

**Review by sameersubudhi — APPROVED** [review]
Thank you for all the changes. LGTM 👍

**Pull request merged**